drivers/include/motor_driver.h

```diff
@@ -143,61 +133,84 @@ typedef enum {
+typedef void (*motor_set_post_cb_t)(const motor_driver_t *motor_driver,
                                   uint8_t motor_id,
                                   int32_t pwm_duty_cycle);
 
+/**
+ * @brief   Motor set callback. Called to set motor speed.
+ */
+typedef void (*motor_set_cb_t)(const motor_t *motor,
+                               motor_direction_t direction);
+
+/**
+ * @brief   Motor brake callback. Called to brake a motor.
+ */
+typedef void (*motor_brake_cb_t)(const motor_t *motor,
+                                 bool brake);
+
 /**
  * @brief Describe DC motor driver with PWM device and motors array
  */
 typedef struct {
-    pwm_t pwm_dev;                          /**< PWM device driving motors */
-    motor_driver_mode_t mode;               /**< driver mode */
-    motor_driver_mode_brake_t mode_brake;   /**< driver brake mode */
-    pwm_mode_t pwm_mode;                    /**< PWM mode */
-    uint32_t pwm_frequency;                 /**< PWM device frequency */
-    uint32_t pwm_resolution;                /**< PWM device resolution */
-    uint8_t nb_motors;                      /**< number of moros */
-    motor_t motors[CONFIG_MOTOR_DRIVER_MAX];       /**< motors array */
-    motor_driver_cb_t cb;                   /**< callback on motor_set */
-} motor_driver_config_t;
+    motor_driver_mode_t mode;                   /**< driver mode */
+    pwm_t pwm_dev;                              /**< PWM device driving motors */
+    pwm_mode_t pwm_mode;                        /**< PWM mode */
+    uint32_t pwm_frequency;                     /**< PWM device frequency */
+    uint32_t pwm_resolution;                    /**< PWM device resolution */
+    bool brake_inverted;                        /**< if false, brake high (1), low (0) otherwise */
+    bool enable_inverted;                       /**< if false, enable high (1), low (0) otherwise */
+    uint8_t nb_motors;                          /**< number of motors */
+    motor_t motors[CONFIG_MOTOR_DRIVER_MAX];    /**< motors array */
+    motor_set_post_cb_t motor_set_post_cb;      /**< callback post to motor_set */
+} motor_driver_params_t;
+
+/**
+ * @brief   Motor driver
+ */
+struct _motor_driver_t {
+    const motor_driver_params_t *params;        /**< parameters */
+};
 
 /**
  * @brief Initialize DC motor driver board
  *
- * @param[out] motor_driver     motor driver to initialize
+ * @param[out]  motor_driver        motor driver to initialize
+ * @param[in]   motor_driver_params motor driver parameters
  *
- * @return                      0 on success
- * @return                      -1 on error with errno set
+ * @return                          0 on success
+ * @return                          -EINVAL on bad parameter value
+ * @return                          -EIO on failed GPIO init
  */
-int motor_driver_init(const motor_driver_t motor_driver);
+int motor_driver_init(motor_driver_t *motor_driver, const motor_driver_params_t *motor_driver_params);
```

Comment:
I think you don't need > 100 chars for this

Reply:
done